### DISCUSSION TITLE
Migrate Data (Course/Uploaded PDF, Videos/Assessments/Resource etc ) from one channel to another

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Knowlg/Community#46] [Q&A] [answered]
**Background**
We have a requirement to migrate data (Course/Uploaded PDF, Videos/Assessments/Resource, etc ) from one channel to another within the same sunbird setup.

Can you please let us know the steps of data migration.

**Approaches -**  
As per our understanding, we have 2 approaches
1. We will write a script - using Read APIs read content from one channel and copy them to another channel using create content APIs.
2. If there is any Bulk Upload script or feature available it will help us.

@amit-tarento Can you please help us.

cc @vinod303, @baljeetsingh, @paritshivani

**Comment by Jayaprakash8887** [comment, marked as answer]
If you want to completely change ownership of the current contents, then org and user related fields (channel, createdFor, createdBy, lastUpdatedBy, lastPublishedBy, etc.) data have to be properly updated with the new Org and user details using "SYSTEM UPDATE API" (non-public API: {{host_pt_kong}}/system/v3/content/update/do_Id). 
If you are ok with dupicate copies of these contents, then as you have mentioned, "copy content API" can be used.

Regarding, bulk upload script, we don't have one. You can create a script which first filters for the desired result of content Id using "composite/content search API" and use the result as input to next set of decided course of action.

**Reply by AmiableAnil** [reply]
@komalm Please let us know if the above answer resolved your issue. Or please mark it as answered
